### PR TITLE
Waiting for response-worker kafka init

### PR DESCRIPTION
**Pull request by slemrmartin**
It seems like first set of response messages are sometimes lost. 

This PR blocks starting thread until kafka listener requests kafka Fetch API.

---

**Dependents**:
- https://github.com/RedHatInsights/topological_inventory-ansible_tower/pull/114

### CODE REVIEW

**Comment by slemrmartin** [comment]
cc @mkanoor

**Comment by slemrmartin** [comment]
@syncrou although the problem we had was somewhere else, this is still needed

**Comment by syncrou** [comment]
@slemrmartin should this be closed for now due to the changes with `ack` defaulting to true?

**Comment by slemrmartin** [comment]
@syncrou it's still valid PR. As the response listener wihout `persist-ref` can't read messages before it was started, 
it can theoretically miss first few responses (with low probability)

File: lib/receptor_controller/client/response_worker.rb
Comment (roliveri) [inline lib/receptor_controller/client/response_worker.rb:79]: @slemrmartin Where is this event published?

Comment (slemrmartin) [reply lib/receptor_controller/client/response_worker.rb:79]: @roliveri they are published by ruby-kafka: https://github.com/zendesk/ruby-kafka/ (chapter Instrumentation in Readme)

File: lib/receptor_controller/client/response_worker.rb
Comment (lindgrenj6) [inline lib/receptor_controller/client/response_worker.rb:58]: I know it's a stretch - but could we maybe have a timeout for this? If the message hasn't been received back for like 15-20 seconds or something it could really indicate we have issues.

File: lib/receptor_controller/client/response_worker.rb
Comment (lindgrenj6) [inline lib/receptor_controller/client/response_worker.rb:79]: I'd like to see a log message here, that way we know it got that far that it's waiting for the client to initialize, maybe something like:

`Initializing Kafka Client...`
then
`Kafka Client successfully initialized.`

---

The current log message doesn't quite makes sense since it isn't a receptor response.

**Review by lindgrenj6 — COMMENTED** [review]
Just a couple small nits, other than that LGTM.

File: lib/receptor_controller/client/response_worker.rb
Comment (roliveri) [reply lib/receptor_controller/client/response_worker.rb:79]: Thanks, that makes things more clear.

**Review by roliveri — APPROVED** [review]
@slemrmartin Let me know if you're going to address Jacob's suggestions. I'll merge once everyone approves.

File: lib/receptor_controller/client/response_worker.rb
Comment (slemrmartin) [reply lib/receptor_controller/client/response_worker.rb:58]: done

**Review by lindgrenj6 — APPROVED** [review]
LGTM!

**Comment by slemrmartin** [comment]
@roliveri Jacob's suggestions implemented

**Pull request merged**
